serial-settings/src/lib.rs

```diff
@@ -0,0 +1,306 @@
+#![no_std]
+
+use core::fmt::Write;
+use miniconf::JsonCoreSlash;
+use embedded_storage::nor_flash::NorFlash;
+use serde::Serialize;
+
+static OUTPUT_BUFFER: bbqueue::BBBuffer<512> = bbqueue::BBBuffer::new();
+
+struct Context<Settings: SerialSettings, Flash: NorFlash + 'static> {
+    output: OutputBuffer,
+    flash: Flash,
+    settings: Settings,
+}
+
+impl<Settings: SerialSettings, Flash: NorFlash + 'static> Context<Settings, Flash> {
+    fn save(&mut self) {
+        self.flash.erase(0, self.flash.capacity() as u32).unwrap();
```

Comment:
No reason to erase all of the flash bank, just erase the area we're going to use.

Reply:
Oh yes.